server/jetty/src/main/java/io/deephaven/server/jetty/jsplugin/JsPlugins.java

```diff
@@ -0,0 +1,105 @@
+/**
+ * Copyright (c) 2016-2022 Deephaven Data Labs and Patent Pending
+ */
+package io.deephaven.server.jetty.jsplugin;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.jimfs.Configuration;
+import com.google.common.jimfs.Jimfs;
+import io.deephaven.plugin.type.JsPlugin;
+import io.deephaven.plugin.type.JsPluginRegistration;
+import org.eclipse.jetty.servlet.DefaultServlet;
+import org.eclipse.jetty.servlet.ServletHolder;
+
+import java.io.BufferedOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.io.UncheckedIOException;
+import java.nio.file.FileSystem;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * A {@link JsPluginRegistration} for Jetty.
+ *
+ * @see #servletHolder(String)
+ */
+public final class JsPlugins implements JsPluginRegistration {
+
+    /**
+     * Creates a new js plugins instance with an in-memory filesystem.
+     *
+     * @return the js plugins
+     * @throws IOException if an I/O exception occurs
+     */
+    public static JsPlugins create() throws IOException {
+        final FileSystem fs = Jimfs.newFileSystem(Configuration.unix());
+        final JsPlugins jsPlugins = new JsPlugins(fs.getPath("/"));
```

Comment:
I had another code path I was testing with earlier - a temporary local directory that was being created. If we add unit testing, we'll probably need to extend our construction techniques to better clean-up.

I'm not sure if this is a personal bias, or recommended best-practices I've seen elsewhere; but I try and keep my constructors as simple setters and simple object creation, with necessary supporting logic and initialization in static methods.

Reply:
That's a good question - I wonder if we are better off choosing the in-memory representation that more closely matches the host operating system. I'll look into it a bit more, specifically wrt `COPY_ATTRIBUTES`

Reply:
I'm not sure we want all attributes, but I'm not sure we don't, just want to be sure we decide rather than get the default. 

Temp dir seems great too, though I'd hesitate about not testing the actual paths we intend to use, since jimfs by design isn't "the real thing", and those oddities might surprise us at runtime.

Thoughts about just making an on-disk zip file of the dir structure we need, and using that as a filesystem? Only one file need be created, and I _believe_ that the zip fs impl holds open a handle, so /tmp being deleted wouldnt hurt us, and we wouldn't need to worry about having tons of and/or giant files in memory.

No objection to simple object creation, just voting for consistency of the class's own API. I tend to agree that if the logic is at all non-trivial, use a nicely named factory, but if there is a simple variant then also give it an appropriately named factory method.

Reply:
I've got zip filesystem plumbed though now, I like it more than jimfs (even though jimfs is cool, it's an extra dependency).